FHEM/00_SIGNALduino.pm

```diff
@@ -2277,11 +2299,19 @@ sub SIGNALduino_Parse_MU($$$$@)
 			{
 				if  (($pstr=SIGNALduino_PatternExists($hash,\@{$ProtocolListSIGNALduino{$id}{zero}},\%patternList,\$rawData)) eq -1)
 				{
-					SIGNALduino_Log3 $name, 5, "$name: zero pattern for MU Protocol id $id not found, aborting";
+					Log3 $name, 5, "$name: zero pattern for MU Protocol id $id not found, aborting" if ($dummy);
 					next;
 				}
 				Debug "Found matched zero" if ($debug);
-
+				if ($clocksource eq "zero")		# clocksource zero, dann die clock aus zero holen
+				{
+					$msgclock = $msg_parts{pattern}{substr($pstr, $ProtocolListSIGNALduino{$id}{clockpos}[1], 1)};
+					$ProtocListClock = $clockabs * $ProtocolListSIGNALduino{$id}{zero}[$ProtocolListSIGNALduino{$id}{clockpos}[1]];
```

Comment:
Der code ist doch identisch mit dem in Zeile 2288 oder?
Der Unterschied ist {zero} oder {one} was in der Variable $clocksource steht.
Der Code könnte doch kombiniert werden, wenn die Zeile so angepasst wird:

`$ProtocListClock = $clockabs * $ProtocolListSIGNALduino{$id}{$clocksource }[$ProtocolListSIGNALduino{$id}{clockpos}[1]];`

Reply:
Welchen Vorteil bringt dies?
Möchstet Du diese Zeile in eine Routine auslagern?

Reply:
Der Vorteil ist, dass wir den Code nur einmal haben.
Das Auslagern in eine eigene SUB wäre durchaus auch gut, da wir diese Funktion dann mit den verschiedenen Parametern testen könnten